koin-projects/koin-android-viewmodel/src/main/java/org/koin/android/viewmodel/ext/android/ActivityExt.kt

```diff
@@ -29,41 +30,38 @@ import kotlin.reflect.KClass
  *
  * @author Arnaud Giuliani
  */
-
-/**
- * LifecycleOwner extensions to help for ViewModel
- *
- * @author Arnaud Giuliani
- */
-
-fun <T : ViewModel> ViewModelStoreOwner.viewModel(
-    clazz: KClass<T>,
+inline fun <reified T : ViewModel> AppCompatActivity.viewModel(
     qualifier: Qualifier? = null,
-    parameters: ParametersDefinition? = null
+    noinline owner: ViewModelOwnerDefinition = { from(this) },
+    noinline parameters: ParametersDefinition? = null
 ): Lazy<T> {
-    return lazy(LazyThreadSafetyMode.NONE) { getViewModel(clazz, qualifier, parameters) }
+    return lazy(LazyThreadSafetyMode.NONE) {
+        getViewModel<T>(qualifier, owner, parameters)
+    }
 }
 
-inline fun <reified T : ViewModel> ViewModelStoreOwner.viewModel(
+fun <T : ViewModel> AppCompatActivity.viewModel(
     qualifier: Qualifier? = null,
-    noinline parameters: ParametersDefinition? = null
+    owner: ViewModelOwnerDefinition = { from(this) },
+    clazz: KClass<T>,
+    parameters: ParametersDefinition? = null
 ): Lazy<T> {
-    return lazy(LazyThreadSafetyMode.NONE) { getViewModel<T>(qualifier, parameters) }
+    return lazy(LazyThreadSafetyMode.NONE) { getViewModel(qualifier, owner, clazz, parameters) }
 }
 
-inline fun <reified T : ViewModel> ViewModelStoreOwner.getViewModel(
+inline fun <reified T : ViewModel> AppCompatActivity.getViewModel(
```

Comment:
Better to use [ComponentActivity](https://developer.android.com/reference/androidx/activity/ComponentActivity)